### PR TITLE
Renamed demos to examples in apex-malhar.

### PR DESCRIPTION
**Pull request by prasannapramod**
@ashwinchandrap @tweise @amberarrow please see.

### CODE REVIEW

**Comment by tweise** [comment]
Glad to see this on the way. I would also like to see the following: 

All packages renamed from com.datatorrrent to org.apache.apex (you are already changing the package name)

Better (consistent) artifactId (like malhar-examples-pi etc.)

**Comment by prasannapramod** [comment]
@ashwinchandrap @tweise @amberarrow I made the necessary changes. Please see.

File: README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
 
-Malhar repository contains open source operator and codec library that can be used with the Apache Apex platform to build Realtime streaming applications. In addition to the library there are benchmark, contrib, demos, webdemos and samples folders available. Demos contain demo applications built using the library operators. Webdemos contain webpages for the demos. Benchmark contains performance testing applications. Contrib contains additional operators that interface with third party softwares. Samples contain some sample code that shows how to use the library operators.
+Malhar repository contains open source operator and codec library that can be used with the Apache Apex platform to build Realtime streaming applications. In addition to the library there are benchmark, contrib, examples, webdemos and samples folders available. Examples contain demo applications built using the library operators. Webdemos contain webpages for the examples. Benchmark contains performance testing applications. Contrib contains additional operators that interface with third party softwares. Samples contain some sample code that shows how to use the library operators.
```
Comment (tweise) [inline README.md]: There is no more webdemos. Please check entire text.

File: examples/twitter/src/main/java/org/apache/apex/examples/twitter/KinesisHashtagsApplication.java [hunk 2]
```diff
@@ -135,12 +135,12 @@
  * <ul>
  * <li><b>The twitterFeed operator : </b> This operator samples random public
  * statues from twitter and emits to application. <br>
- * Class : com.datatorrent.demos.twitter.TwitterSampleInput <br>
+ * Class : com.datatorrent.examples.twitter.TwitterSampleInput <br>
```
Comment (tweise) [inline examples/twitter/src/main/java/org/apache/apex/examples/twitter/KinesisHashtagsApplication.java]: Package changed.

**Review by tweise — APPROVED** [review]
LGTM.

Couple of text changes that could be addressed as follow-up as well. Would also be good to remove obsolete examples such as frauddetect as follow-up work.

File: examples/highlevelapi/pom.xml [hunk 3]
```diff
@@ -22,15 +22,15 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
 
-  <artifactId>high-level-api-demo</artifactId>
+  <artifactId>malhar-examples-high-level-api</artifactId>
```
Comment (tweise) [inline examples/highlevelapi/pom.xml]: high-level-api -> highlevelapi

**Comment by amberarrow** [comment]
There are still several occurrences of 'demo'; you'll see them with "git grep demo"

**Comment by prasannapramod** [comment]
@amberarrow @tweise @ashwinchandrap addressed all the comments. please review.

File: README.md [hunk 4]
```diff
@@ -1,7 +1,7 @@
 Apache Apex Malhar
 ===============================
 
-Malhar repository contains open source operator and codec library that can be used with the Apache Apex platform to build Realtime streaming applications. In addition to the library there are benchmark, contrib, demos, webdemos and samples folders available. Demos contain demo applications built using the library operators. Webdemos contain webpages for the demos. Benchmark contains performance testing applications. Contrib contains additional operators that interface with third party softwares. Samples contain some sample code that shows how to use the library operators.
+Malhar repository contains open source operator and codec library that can be used with the Apache Apex platform to build Realtime streaming applications. In addition to the library there are benchmark, contrib, examples and samples folders available. Examples contain applications built using the library operators. Benchmark contains performance testing applications. Contrib contains additional operators that interface with third party softwares. Samples contain some sample code that shows how to use the library operators.
```
Comment (tweise) [inline README.md:4]: Please create a follow-up, separate PR to get rid of samples, they are obsolete.

**Comment by tweise** [comment]
Can you please rebase:
```
git rebase apache/master
```
Thanks

**Comment by tweise** [comment]
LGTM, tests pass. Will merge as soon as the rebase is done.

**Comment by prasannapramod** [comment]
rebased

File: README.md (hunk 4 above)
Comment (prasannapramod) [reply README.md:4]: I will look into samples in the next iteration.

**Comment by tweise** [comment]
Merged, it's just not showing up on the mirror yet.

**Pull request merged**